File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.cs
public class Solution {
    //time - O(n^2)
    //space - O(1)
    public string LongestPalindrome(string s) {
        int left = 0;
        int maxLength = 0;

        for(int i = 0; i< s.Length; i++) {
            int oddStringLength = isPalindrome(i, i, s);
            int evenStringLength = isPalindrome(i, i+1, s);
            int tmpMaxLength = Math.Max(oddStringLength, evenStringLength);
            if(tmpMaxLength > maxLength) {
                left = i - ((tmpMaxLength - 1) / 2);
                maxLength = tmpMaxLength;
            }
        }

        return s.Substring(left, maxLength);
    }

    private int isPalindrome(int start, int end, string s) {
        while(start >= 0 && end < s.Length && s[start] == s[end]) {
            start--;
            end++;
        }
 
        return end - start - 1;
    }
}

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        '''
        even and odd expansion basically
        - function to do palindrom check
        '''
        left = 0
        maxLength = 0
        
        for i in range(len(s)):
            maxEvenLength = self.isPalindrome(i, i, s)
            maxOddLength = self.isPalindrome(i, i+1, s)
            maxTempLength = max(maxEvenLength, maxOddLength)
            if maxTempLength > maxLength:
                left = i - ((maxTempLength - 1) // 2)
                maxLength = maxTempLength
                
        return s[left: left+maxLength]
        
            
    def isPalindrome(self, left, right, s):
        while left >= 0 and right < len(s) and s[left] == s[right]:
            left -= 1
            right += 1
            
        return right - left - 1